File: db/csharp/SalesManagement.Domain/Models/PurchaseDetail.cs
namespace SalesManagement.Domain.Models
{
    /// <summary>
    /// 仕入データ明細
    /// </summary>
    public class PurchaseDetail
    {
        public string PurchaseNo { get; set; } = string.Empty;
        public int PurchaseRowNo { get; set; }
        public string ProductCode { get; set; } = string.Empty;
        public string ProductName { get; set; } = string.Empty;
        public int UnitPrice { get; set; }
        public int Quantity { get; set; }
        public int ConsumptionTaxRate { get; set; }
        public string LotNo { get; set; } = string.Empty;
        public string WarehouseCode { get; set; } = string.Empty;
        public int Discount { get; set; }
        public DateTime CreatedAt { get; set; }
        public string CreatedBy { get; set; } = string.Empty;
        public DateTime UpdatedAt { get; set; }
        public string UpdatedBy { get; set; } = string.Empty;
    }
}

File: db/fsharp/SalesManagement.Domain/Models/PurchaseDetail.fs
namespace SalesManagement.Domain.Models

open System

/// <summary>
/// 仕入データ明細
/// </summary>
[<CLIMutable>]
type PurchaseDetail = {
    PurchaseNo: string             // 仕入番号（PK、FK）
    PurchaseLineNo: int            // 仕入行番号（PK）
    ProductCode: string            // 商品コード（FK）
    ProductName: string            // 商品名（履歴保持）
    LotNo: string                  // ロット番号
    WarehouseCode: string          // 倉庫コード（FK）
    PurchaseQuantity: int          // 仕入数量
    UnitPrice: int                 // 仕入単価（履歴保持）
    Amount: int                    // 金額
    ConsumptionTax: int            // 消費税
    CreatedAt: DateTime            // 作成日時
    CreatedBy: string              // 作成者名
    UpdatedAt: DateTime            // 更新日時
    UpdatedBy: string              // 更新者名
}
